### PR TITLE
Profile Backend

### PR DESCRIPTION
**Pull request by AbdallahAmmar96**
- Edit the app.js file to handle the static and controllers
- make the test for the /api/v1/student/user/id
- edit the file structure in the controller
- add the supertest

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
 const app = express();
 app.disable('x-powered-by');
 app.use(express.json());
 app.set('port', process.env.PORT || 7777);
 
+app.use(express.static(join(__dirname, '..', 'Client', 'build')));
+app.use('/api/v1', router);
+
+app.get('*', (req, res) => {
+  res.sendFile(join(__dirname, '..', 'CLient', 'build', 'index.html'));
```
Comment (kannanhassouna) [inline server/app.js]:
```suggestion
  res.sendFile(join(__dirname, '..', 'Client', 'build', 'index.html'));
```

File: server/controllers/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const router = require('express').Router();
+const student = require('./student');
+// const staff = require('./staff');
```
Comment (kannanhassouna) [inline server/controllers/index.js]: remove it until you use it

File: server/controllers/student/user/get.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const getStudentInformation = require('../../../database/queries/getStudent');
+
+exports.get = (req, res) => {
+  getStudentInformation('12345')
```
Comment (kannanhassouna) [inline server/controllers/student/user/get.js]:
```suggestion
const { id } = req.params;
  getStudentInformation(id)
```

File: server/app.js [hunk 4]
```diff
@@ -1,8 +1,17 @@
 const express = require('express');
+const { join } = require('path');
+const router = require('./controllers');
 
 const app = express();
 app.disable('x-powered-by');
 app.use(express.json());
 app.set('port', process.env.PORT || 7777);
```
Comment (kannanhassouna) [inline server/app.js]:
```suggestion
app.use(express.urlencoded({ extends: false }));	
```

File: server/controllers/index.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+router.use('/student', student);
+// router.use('/staff', staff);
```
Comment (kannanhassouna) [inline server/controllers/index.js]: you can remove it, i got your idea

File: server/controllers/student/courses/index.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const router = require('express').Router();
+const { getAll } = require('./getAll');
+const { getmy } = require('./getMy');
+const { post } = require('./postApply');
+
+router.route('/allcourses')
+  .get(getAll);
+
+router.route('/mycourses')
+  .get(getmy);
```
Comment (kannanhassouna) [inline server/controllers/student/courses/index.js]:
```suggestion
  .get(studentCourses);
```

File: server/controllers/student/courses/index.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+const router = require('express').Router();
+const { getAll } = require('./getAll');
+const { getmy } = require('./getMy');
+const { post } = require('./postApply');
+
+router.route('/allcourses')
+  .get(getAll);
```
Comment (kannanhassouna) [inline server/controllers/student/courses/index.js]:
```suggestion
  .get(getCourses);
```

File: server/controllers/student/courses/index.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+router.route('/mycourses')
+  .get(getmy);
+
+router.route('/applycourse')
+  .post(post);
```
Comment (kannanhassouna) [inline server/controllers/student/courses/index.js]:
```suggestion
  .post(applyToCourse);
```

File: server/controllers/student/user/get.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    .then((information) => {
+      res.send({
+        error: null,
+        data: information.rows,
+      });
+    })
+    .catch(err => res.send({
+      error: err,
```
Comment (kannanhassouna) [inline server/controllers/student/user/get.js]: you can't send the error as you got it, you should handle it 
```suggestion
      error: ' Sorry we have some issues',
```

File: server/test/Endpoints/router-test.js [hunk 10]
```diff
@@ -1,6 +1,29 @@
...
+const dbBuild = require('../../database/config/db_build');
 
-tape('just for testing', (t) => {
-  t.equal(1, 1, 'on is equal one');
-  t.end();
+test('Get student information from /api/v1/student/user/12345', (t) => {
+  dbBuild()
+    .then(() => {
+      request(app)
+        .get('/api/v1/student/user/12345')
```
Comment (kannanhassouna) [inline server/test/Endpoints/router-test.js:10]: you should try the Abnormal path

File: server/test/Endpoints/router-test.js [hunk 11]
```diff
@@ -1,6 +1,29 @@
...
+        .expect(200)
+        .expect('Content-Type', /json/)
+        .end((err, res) => {
+          const obj = JSON.parse(res.text);
+          if (err) {
+            t.error(err);
+          } else if (obj.error) {
+            t.error(obj.error);
```
Comment (kannanhassouna) [inline server/test/Endpoints/router-test.js]: No  No No you should check the error if it as you sent it from backend. 
```suggestion
            t.equal(err.error, 'Sorry we have some issues', 'get same data that expected');
```

**Review by kannanhassouna — CHANGES_REQUESTED** [review]

File: server/app.js (hunk 4 above)
Comment (AbdallahAmmar96) [reply server/app.js]: you mean
```suggestion
app.use(express.urlencoded({ extended: false }));
```

**Review by kannanhassouna — APPROVED** [review]

**Pull request merged**